share/picongpu/tests/XrayScattering/include/picongpu/param/speciesDefinition.param

```diff
@@ -0,0 +1,133 @@
+ * `VectorAllSpecies`.
+ */
+
+#pragma once
+
+#include "picongpu/simulation_defines.hpp"
+#include "picongpu/particles/Particles.hpp"
+
+#include <pmacc/particles/Identifier.hpp>
+#include <pmacc/meta/conversion/MakeSeq.hpp>
+#include <pmacc/identifier/value_identifier.hpp>
+#include <pmacc/particles/traits/FilterByFlag.hpp>
+#include <pmacc/meta/String.hpp>
+
+namespace picongpu
+{
+
+/*########################### define particle attributes #####################*/
+
+/** describe attributes of a particle*/
+using DefaultParticleAttributes = MakeSeq_t<
+    position< position_pic >,
+    momentum,
+    weighting
+>;
+
+/*########################### end particle attributes ########################*/
+
+/*########################### define species #################################*/
+
+/*--------------------------- electrons --------------------------------------*/
+
+/* ratio relative to BASE_CHARGE and BASE_MASS */
+value_identifier( float_X, MassRatioElectrons, 1.0 );
+value_identifier( float_X, ChargeRatioElectrons, 1.0 );
+
+using ParticleFlagsElectrons = MakeSeq_t<
+    particlePusher< UsedParticlePusher >,
+    shape< UsedParticleShape >,
+    interpolation< UsedField2Particle >,
+    current< UsedParticleCurrentSolver >,
+    massRatio< MassRatioElectrons >,
+    chargeRatio< ChargeRatioElectrons >
+#if( ENABLE_SYNCHROTRON_PHOTONS == 1 )
+    , synchrotronPhotons< PIC_Photons >
+#endif
+>;
+
+/* define species electrons */
+using PIC_Electrons = Particles<
+    PMACC_CSTRING( "e" ),
+    ParticleFlagsElectrons,
+    DefaultParticleAttributes
+>;
+
+/*--------------------------- ions -------------------------------------------*/
+
+/* ratio relative to BASE_CHARGE and BASE_MASS */
+value_identifier( float_X, MassRatioIons, 115837 );
+value_identifier( float_X, ChargeRatioIons, -29.0 );
+
+/* ratio relative to BASE_DENSITY */
+value_identifier( float_X, DensityRatioIons, 1.0 );
+
+using ParticleFlagsIons = MakeSeq_t<
+    particlePusher< UsedParticlePusher >,
+    shape< UsedParticleShape >,
+    interpolation< UsedField2Particle >,
+    current< UsedParticleCurrentSolver >,
+    massRatio< MassRatioIons >,
+    chargeRatio< ChargeRatioIons >,
+    densityRatio< DensityRatioIons >,
+    atomicNumbers< ionization::atomicNumbers::Copper_t >
+>;
+
+/* define species ions */
+using PIC_Ions = Particles<
+    PMACC_CSTRING( "i" ),
+    ParticleFlagsIons,
+    MakeSeq_t<
+        DefaultParticleAttributes,
+        boundElectrons >
+>;
+
+/*########################### end species ####################################*/
+
+/*enable (1) or disable (0) ions*/
+#ifndef PARAM_IONS
+#   define PARAM_IONS 0
+#endif
+
+/** All known particle species of the simulation
+ *
+ * List all defined particle species from above in this list
+ * to make them available to the PIC algorithm.
+ */
+using VectorAllSpecies = MakeSeq_t<
+    PIC_Electrons
+#if( PARAM_IONS == 1)
+    ,PIC_Ions
+#endif
+    >;
+
+} // namespace picongpu
```

Comment:
Is there a difference to the default case, if not please remove the file?

Reply:
I'm adding here an option for PIC_Ions with `boundElectrons`. Also I'm setting the ion charge to sth different than 1.